docs/developer/available-plugins/adyen.mdx

```diff
@@ -0,0 +1,269 @@
+ As a next step, Saleor will use `returnUrl` provided in `checkoutPaymentCreate` mutation to redirect shopper to store page.
+The `returnUrl` will have attached below parameters:
+  - Required:
+    - `checkout` - the ID of processed checkout.
+    - `payment` - the ID of processed payment.
+    - `resultCode` - status of the payment recieved from Adyen. Can be used to determine the current status of the payment. [Docs](https://docs.adyen.com/online-payments/payment-result-codes)
+  - Optional:
+    - `action` -  action data if an additional action is required by customer.
+
+If payment was successfully processed, Saleor will create an order. The next mutation `checkoutComplete` call will return `order` in the response.
+
+
+
+## Enabling native 3DS2
+Saleor by default uses redirect 3D Secure 1 and 2. To use native 3DS, [enable it in the Saleor's dashboard](dashboard/configuration/plugins/adyen.md#enabling-the-adyen-integration). 
+Setting this flag will include to Adyen's request `additionalData.allow3DS2: true`, to all card transactions.
+Make sure that you provide all required fields for native 3D Secure in `checkoutComplete.input.paymentData` as described [here](developer/available-plugins/adyen.mdx#completing-the-checkout). 
+The fields mandatory for enabling native 3D Secure 2 can be found [here](https://docs.adyen.com/online-payments/3d-secure/native-3ds2). 
+Specific channel (`iOS`, `Android`, `Web`) or solution type (`Drop-in`, `Components`) can require different fields. 
+
+
+## Enabling Apple Pay
+
+:::note
+Make sure that you follow all steps required to activate apple pay as described [here](dashboard/configuration/plugins/adyen.md#activating-applePay)
+:::
+
+### Enabling on web channel
+Each usage Apple Pay on web requires the initialization of the payment session.
+Adyen's form([Drop-in](https://docs.adyen.com/payment-methods/apple-pay/web-drop-in?tab=_code_payments_code__2#drop-in-configuration), 
+[Components](https://docs.adyen.com/payment-methods/apple-pay/web-component?tab=_code_payments_code__2#step-2-create-an-instance-of-the-component)) has handler `onValidateMerchant`. To request the Apple Pay session, 
+call mutation `paymentInitialize` inside `onValidateMerchant`.
+
+
+Mutation `paymentInitialize` accepts below fields:
+- `channel`: channel slug, where the Apple Pay session should be initialized
+- `gateway`: - The `gateway` is the id of the  plugin which for Adyen is  - `mirumee.payments.adyen`
+- `paymentData`: JSONString with all data required to initialize session.
+
+`PaymentData` for initializing Apple Pay session should contain the below structure.
+```json
+{
+  "merchantIdentifier": <your-apple-merchant-id>,
+  "displayName": "Merchant ID for Saleor and Adyen",
+  "domain": "The registered storefront domain",
+  "validationUrl": <validationUrl>,
+  "paymentMethod": "applepay"
+}
```

Comment:
```suggestion
{
  "merchantIdentifier": "<your-apple-merchant-id>",
  "displayName": "Merchant ID for Saleor and Adyen",
  "domain": "The registered storefront domain",
  "validationUrl": "<validationUrl>",
  "paymentMethod": "applepay"
}
```

Make it a valid json to avoid formatting issues